src/Microsoft.AspNetCore.Mvc.Core/Authorization/AuthorizeFilter.cs

```diff
@@ -46,6 +77,12 @@ public AuthorizeFilter(AuthorizationPolicy policy)
                 throw new ArgumentNullException(nameof(context));
             }
 
+            Policy = Policy ?? await AuthorizationPolicy.CombineAsync(PolicyProvider, AuthorizeData);
+            if (Policy == null)
+            {
+                return;
```

Comment:
I think there the idea. Always calling Into the provider since we don't know when combine will start returning different policy data for the same inputs.